spec/models/google_analytics/data_tracking_spec.rb

```diff
@@ -2,22 +2,36 @@
 
 module GoogleAnalytics
   describe DataTracking do
+    before do
+      allow(described_class).to receive(:adapter).and_return('Adapter')
+      allow(Rails).to receive(:env).and_return('production'.inquiry)
+      described_class.active = true
+    end
+
     describe '#enabled?' do
       subject { described_class.enabled? }
-      before do
-        allow(Rails).to receive(:env).and_return('production'.inquiry)
-      end
 
-      context 'with an adapter set' do
-        before do
-          allow(described_class).to receive(:adapter).and_return('Adapter')
+      context 'when host is staging' do
+        before { allow(RailsHost).to receive(:env).and_return('staging') }
```

Comment:
need to check this ENV['ENV'] is Rails.env....

Reply:
ok, this is just a refactor of the existing tests I see now, breaking the loop down. The diff is a bit confusing.